hassrelease/changelog.py

```diff
@@ -107,52 +104,49 @@ def generate(release, prs, *, website_tags):
 
     changes = []
     links = set()
-    for line in release.log_lines():
-        labels = []
-        parts = ['-', line.message]
-
-        if line.email not in users:
-            print('Error! Found unresolved user', line.email)
-            sys.exit(1)
+    with click.progressbar(release.log_lines(), label="Changelog") as bar:
+        for line in bar:
+            parts = ['-', line.message]
 
-        # Filter out git commits that are not merge commits
-        if line.pr is None:
-            continue
-
-        pr = prs.get(line.pr)
+            # Filter out git commits that are not merge commits
+            if line.pr is None:
+                continue
 
-        if (pr.milestone is not None and
-            StrictVersion(pr.milestone.title).version !=
-                release.version.version):  # Ignore beta version tag
-            continue
+            pr = prs.get(line.pr)
+    
+            if (pr.milestone is not None and
+                StrictVersion(pr.milestone.title).version !=
+                    release.version.version):  # Ignore beta version tag
+                continue
 
-        labels = [label.name for label in pr.labels()]
+            labels = [label.name for label in pr.labels()]
 
-        # Filter out commits for which the PR has one of the ignored labels
-        if any(label in IGNORE_LINE_LABELS for label in labels):
-            continue
+            # Filter out commits for which the PR has one of the ignored labels
+            if any(label in IGNORE_LINE_LABELS for label in labels):
+                continue
 
-        links.add(LINK_DEF_USER.format(users[line.email]))
-        parts.append(INFO_TEMPLATE.format(users[line.email], line.pr))
-        links.add(LINK_DEF_PR.format(line.pr))
+            user = pr.user
+            links.add(LINK_DEF_USER.format(user.login, user.html_url))
+            parts.append(INFO_TEMPLATE.format(user.login, pr.number))
+            links.add(LINK_DEF_PR.format(pr.number, pr.html_url))
```

Comment:
I was always wondering about these lines: Why are reference-style links used here? I mean the logic would get a bit simpler if we would just use inline links everywhere... Is it maybe a performance thing for jekyll?

Reply:
Just to keep things smaller, if 1 user adds 10 PRs, it's now only included once (`links` is a set)